src/sp_lev.c

```diff
@@ -2078,6 +2078,12 @@ struct mkroom *croom;
             discard_minvent(mtmp, TRUE);
             invent_carrying_monster = mtmp;
         }
+        if (m->dead) {
+            mondied(mtmp);
+            /* kludge for this: it didn't actually die while the player was
+             * around, so revert mondead() incrementing this */
+            mvitals[monsndx(mtmp->data)].died--;
```

Comment:
For the same reason that decrementing the `died` count for that monster in `g.mvitals` is necessary -- i.e. a monster marked "dead" is created and then killed, with all the normal side-effects of a monster dying in-game -- when you are playing as a Cartomancer monsters marked "dead" can still drop monster cards as usual.  They will also drop corpses at the normal rate, so there's no guarantee of getting a corpse from a particular "dead" monster, which could be a problem for your use case (trying to ensure there is a corpse called "chef" with various cooking implements, right?).

I think you can add `&& !g.in_mklev` to the [conditions associated with creating a monster card](https://github.com/NullCGT/SpliceHack/blob/b7bef9595ee5d0d1c6ca39f51506515465ef99d4/src/mon.c#L2782-L2788) and add `|| g.in_mklev` to the [conditions in which a corpse is always created](https://github.com/NullCGT/SpliceHack/blob/b7bef9595ee5d0d1c6ca39f51506515465ef99d4/src/mon.c#L3043-L3045) to force each "dead" monster to leave a corpse but not a monster card (you could probably also do something similar [here](https://github.com/NullCGT/SpliceHack/blob/b7bef9595ee5d0d1c6ca39f51506515465ef99d4/src/mon.c#L2839) to ensure the `died` count is only incremented if `!g.in_mklev`, and then you wouldn't need to decrement it here).

Reply:
hrrrmmm.  thanks for pointing that out.  it's weird for monsters who don't die at the player's hands to still drop a card.  like, Archons whacking each other due to Conflict and you can just grab a card off the corpse, lol.

but that's a whole nother PR.  first, i'm gonna try and figure this PR out soon™️, pretty busy IRL and not good at coding!

and no, actually i want chef to be alive - it's the shks and watch who should be dead. the chef works for the orcs 😁

Reply:
I would guess the reason monsters drop a card when they die for any reason, even without the player's direct involvement, is to facilitate the Cartomancer being used as a sort of pet-wrangling pacifist character -- since the way the role plays is based around collecting cards and then summoning monsters to attack on your behalf, it does make sense to me that card-dropping doesn't rely on the player being the one who killed the monster. 

I think it would be worth considering making  dropping a card and dropping a corpse mutually exclusive as part of exploring potential further restrictions (e.g. cards only drop if killed by a tame mon or the hero) since this could lead to some interesting trade-offs between using a monster card and getting a lower chance of food/intrinsics, but potentially more cards, vs using conflict or something and getting potentially more food/intrinsics but no more cards. That might inject some interesting tactical decisions into pacifist Cartomancer gameplay.

But like you said, this is not really relevant to this PR. Just thinking about it!

> and no, actually i want chef to be alive - it's the shks and watch who should be dead. the chef works for the orcs 😁

Oh yeah, sorry about that -- I should have referred back to the other conversation before commenting.